### PR TITLE
python3Packages.ansible: 2.8.4 ->  2.9.2

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
noticed @r-ryantm couldn't update it

closes  #70326
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

failures are broken on master
```
[3 built, 1 copied (10.2 MiB), 10.1 MiB DL]
https://github.com/NixOS/nixpkgs/pull/72844
3 package were built:
ansible ansible_2_6 ansible_2_7
```

### CODE REVIEW

**Comment by jonringer** [comment]
@GrahamcOfBorg build ansible kargo python27Packages.ansible python27Packages.ansible-runner python27Packages.pytest-ansible python37Packages.ansible-kernel python37Packages.ansible-runner python37Packages.pytest-ansible python38Packages.ansible python38Packages.ansible-kernel python38Packages.ansible-runner python38Packages.pytest-ansible

**Review by c0bw3b — DISMISSED** [review]
Careful here, Ansible minor versions introduce breaking changes and require [some porting](https://docs.ansible.com/ansible/latest/porting_guides/porting_guide_2.9.html).
That's why we have `ansible_2_6`, `ansible_2_7` and `ansible` = Ansible 2.8

OTOH there is a [v2.8.**6**](https://github.com/ansible/ansible/blob/stable-2.8/changelogs/CHANGELOG-v2.8.rst) available and it contains CVE fixes too. So it should probably be updated to that version and backported.
\+ same CVE fixes in [v2.6.20](https://github.com/ansible/ansible/blob/stable-2.6/changelogs/CHANGELOG-v2.6.rst) and [v2.7.14](https://github.com/ansible/ansible/blob/stable-2.7/changelogs/CHANGELOG-v2.7.rst)

**Comment by jonringer** [comment]
i could freeze the ansible package to 2.8, but what i mean to change was just the python package.

**Comment by jonringer** [comment]
why would ansible release a minor version with breaking changes....

**Comment by c0bw3b** [comment]
Han >;< it's getting late for me, just saw "ansible: 2.8.x -> 2.9" and thought woops
But I guess the Py package works best with the corresponding Ansible minor version.

> why would ansible release a minor version with breaking changes....
IKR! It's quite painful to work with.. :/

**Comment by c0bw3b** [comment]
Ah still [we define](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/admin/ansible/default.nix#L6):
```nim
  ansible_2_8 = with python3Packages; toPythonApplication ansible;
```

So this bump would actually bump `ansible_2_8` to 2.9.0

**Comment by jonringer** [comment]
yea, I will freeze the current version, and just add an ansible_2_9 entry, then people can opt into it.

**Comment by c0bw3b** [comment]
Yes it's the most practical way.

**Comment by jonringer** [comment]
wondering if i should break this out into two commits, one for freezing, then one for the bump. But I made it so that `ansible` and `ansible_2_8` reference 2_8, and now the python package is at the latest

**Comment by jonringer** [comment]
actually, I bumped ansibled from 2.8.4 to 2.8.6, i should do a separate commit....

File: pkgs/tools/admin/ansible/default.nix [hunk 1]
```diff
@@ -1,9 +1,27 @@
 { python3Packages, fetchurl }:
 
 {
-  ansible = with python3Packages; toPythonApplication ansible;
+  ansible_2_9 = with python3Packages; toPythonApplication ansible;
 
-  ansible_2_8 = with python3Packages; toPythonApplication ansible;
+  ansible = with python3Packages; toPythonApplication (ansible.overridePythonAttrs(old: rec {
```
Comment (Lassulus) [inline pkgs/tools/admin/ansible/default.nix]: why don't we use an recursive attribute-set and say ansible = ansible_2_8; ?

Comment (JohnAZoidberg) [reply pkgs/tools/admin/ansible/default.nix]: bump @jonringer

**Comment by jonringer** [comment]
@GrahamcOfBorg build ansible ansible_2_6 ansible_2_7 ansible_2_9

**Comment by jonringer** [comment]
@GrahamcOfBorg build ansible ansible_2_6 ansible_2_7 ansible_2_9

**Review by c0bw3b — APPROVED** [review]

**Comment by jonringer** [comment]
interesting.... darwin is getting different sha's

**Comment by jonringer** [comment]
cc @Infinisil would you know why darwin is getting different hashes on a .tar.gz?

**Comment by jonringer** [comment]
this is just for ansible_2_9, which means ansible still works on darwin. I might just push :/

**Comment by jonringer** [comment]
@GrahamcOfBorg build ansible_2_9

**Comment by Infinisil** [comment]
The only thing I can think of is filepath unicode normalization, which certain filesystems on macOS do

**Pull request merged**